Spatial_Engine/Query/PointLayout.cs

```diff
@@ -0,0 +1,457 @@
+using BH.oM.Reflection.Attributes;
+using BH.oM.Base;
+using BH.oM.Geometry;
+using BH.oM.Spatial.Layouts;
+using BH.Engine.Geometry;
+
+namespace BH.Engine.Spatial
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Queries the points from the Layout based on the layout and a single region curve associated with it.")]
+        [Input("layout2D", "The layout object to query the points from.")]
+        [Input("hostRegionCurve", "The region curve of the objects associated with the layout.")]
+        [Input("openingCurves", "Optional opening curves in the region.")]
+        [Output("points", "Point layout generated by the layout objects and region curve.")]
+        public static List<Point> PointLayout(this ILayout2D layout2D, ICurve hostRegionCurve, IEnumerable<ICurve> openingCurves = null)
+        {
+            return IPointLayout(layout2D, new List<ICurve> { hostRegionCurve }, openingCurves);
+        }
+
+        /***************************************************/
+
+        [Description("Returns the points in the layout. For ExplicitLayouts, the host geometry will be ignored.")]
+        [Input("layout2D", "The layout object to query the points from.")]
+        [Input("hostRegionCurves", "The region curves of the objects associated with the layout. Unused for Explicit layouts.")]
+        [Input("openingCurves", "Optional opening curves in the region. Unused for ExplicitLayout.")]
+        [Output("points", "The Points stored in the ExplicitLayout object.")]
+        public static List<Point> PointLayout(this ExplicitLayout layout2D, IEnumerable<ICurve> hostRegionCurves = null, IEnumerable<ICurve> openingCurves = null)
+        {
+            return layout2D.Points.ToList();
+        }
+
+        /***************************************************/
+
+        [Description("Returns a series of points along the host region curves, based on the parameters in the PerimeterLayout.")]
+        [Input("layout2D", "The layout object to query the points from.")]
+        [Input("hostRegionCurves", "The region curves of the objects associated with the layout.")]
+        [Input("openingCurves", "Optional opening curves in the region. Unused for PerimeterLayout.")]
+        [Output("points", "Point layout generated by the layout objects and region curve.")]
+        public static List<Point> PointLayout(this PerimeterLayout layout2D, IEnumerable<ICurve> hostRegionCurves, IEnumerable<ICurve> openingCurves = null)
+        {
+            List<ICurve> regionCurveList = hostRegionCurves.ToList();
+            List<int> divs = new List<int>();
+            if (regionCurveList.Count == 1)
+                divs.Add(layout2D.NumberOfPoints);
+            else
+                divs = DistributeDivisions(regionCurveList, layout2D.NumberOfPoints);
```

Comment:
The divisions per curve are calculated before the discontinuity points are considered, meaning all divisions can go to the bigger curve, which won't have enough divisions to make up the difference.

Here I asked for 10 points and got 17, (extreme case)
![image](https://user-images.githubusercontent.com/53530468/80497474-699e3e00-8962-11ea-9e29-851fe8b3e970.png)

Reply:
This was actually intentional. Considered whether to check fro number of points before/after and went with before to as it ensures long curves get more points. If checking after, a large circle in relation to a small polyline with multiple point would give kind of strange results.